src/setup/destroy.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  DeleteBucketCommand,
+  DeleteObjectCommand,
+  ListObjectsCommand,
+} from "@stedi/sdk-client-buckets";
+import { bucketClient } from "../lib/buckets.js";
+import { guidesClient } from "../support/guide.js";
+import { stashClient } from "../lib/stash.js";
+import { DeleteGuideCommand } from "@stedi/sdk-client-guides";
+import {
+  PARTNERS_KEYSPACE_NAME,
+  CONTROL_NUMBER_KEYSPACE_NAME,
+} from "../lib/constants.js";
+import { DeleteKeyspaceCommand } from "@stedi/sdk-client-stash";
+import * as dotenv from "dotenv";
+import { DeleteUserCommand } from "@stedi/sdk-client-sftp";
+import { sftpClient } from "../lib/sftp.js";
+import { functionClient } from "../support/functions.js";
+import { DeleteFunctionCommand } from "@stedi/sdk-client-functions";
+import { updateDotEnvFile } from "../support/utils.js";
+dotenv.config();
+
+(async () => {
+  console.log("Deleting all resources provisioned by bootstrap");
+
+  // Delete Buckets
+  console.log("Deleting Buckets");
+  await emptyAndDeleteBucket(process.env.SFTP_BUCKET_NAME ?? "");
+  await emptyAndDeleteBucket(process.env.EXECUTIONS_BUCKET_NAME ?? "");
+
+  // Delete Guides
+  console.log("Deleting Guides");
+  const guideIds = process.env.GUIDE_IDS?.split(",");
+  for (const guideId of guideIds ?? []) {
+    await guidesClient().send(
+      new DeleteGuideCommand({ id: `LIVE_${guideId}` })
+    );
+    await guidesClient().send(
+      new DeleteGuideCommand({ id: `DRFT_${guideId}` })
+    );
+  }
+
+  // Delete Stash keyspaces
+  console.log("Deleting Stash Keyspaces");
+  await stashClient().send(
+    new DeleteKeyspaceCommand({ keyspaceName: PARTNERS_KEYSPACE_NAME })
```

Comment:
something worth mentioning about the stash apis is that some of the commands are async (like deleting a keyspace). if you want to ensure that it is actually deleting, the status has to be polled. we do that when creating the keyspaces. 

if deleting these resources is just intended to be best-effort, the current implementation should be fine. if we want to let users know if/when it doesn't delete properly, we have to check the status

Reply:
💯